### ISSUE TITLE
[Feature/BugFix] PathPrefix

### ISSUE THREAD

**Issue by ItachiSan** [pyload/pyload#1128]
As in title (I saw some non-merged commits but no general issue, so I created this; maybe others could be referenced here)

**What is expected**
Using a reverse proxy configuration on a webserver plus the path prefix option in the webinterface, any user should be able to set its pyLoad WebUI to a URL like _proto://my_server/my_prefix_
**The reality**
As for now, the only possible reverse proxy setup is in the main directory (like: _proto://my_server/_)
**What I whish from Santa**
The expected version.

**Comment by I-RiX-I** [comment]
came here for the same issue. rebinding the path prefix does not in fact change the behavior of pyload behind an apache reverse proxy.

**Comment by I-RiX-I** [comment]
This should get us there. Needs a fixup, though
https://github.com/pyload/pyload/pull/2015

**Comment by DexterHovis9000** [comment]
here is my workaround for an apache proxy with url like "https://MYHOST/pyload"

prerequisites: already setup apache with ssl

1. enable SSL inside pyload with local generated ssl certificate
2. enable apache mods: mod_ssl, mod_proxy, mod_proxy_http, mod_proxy_html, mod_substitute, mod_headers
3. here is my pyload apache configuration

```
SSLProxyEngine On
SSLProxyVerify none
SSLProxyCheckPeerCN off
SSLProxyCheckPeerName off
SSLProxyCheckPeerExpire off

<Location /pyload>
	SSLRequireSSL
	ProxyPass https://localhost:8000/pyload
	ProxyPassReverse https://localhost:8000/pyload
	ProxyPreserveHost On
	ProxyHTMLEnable On
	ProxyHTMLCharsetOut *
	ProxyHTMLURLMap / /pyload/
	SetOutputFilter INFLATE;DEFLATE
	AddOutputFilterByType INFLATE;SUBSTITUTE;DEFLATE text/css
	AddOutputFilterByType INFLATE;SUBSTITUTE;DEFLATE text/javascript
	Substitute "s|/media|/pyload/media|i"
	Substitute "s|/json|/pyload/json|i"
	Substitute "s|/api|/pyload/api|i"
	Header edit Location ^https://MYHOST/(?!pyload\/)(.*)$ https://MYHOST/pyload/$
</Location>
```

explanation:
first lines disable ssl verification check for local connection between apache and pyload (since this is typically self signed).
inside location, there is setup a typical reverse proxy.
mod_html is used to replace all html url from the webui to /pyload/
mod_substitute is used to replace all javascript/css urls to /pyload/, too
mod_headers is used to correct http redirects (ie. to the login page)

for debian/ubuntu users: you need to add something like this [proxy_html.conf](http://apache.webthing.com/svn/apache/filters/proxy_html/proxy_html.conf) to your (global) apache configuration.

**Comment by oaigh** [comment]
Hi,

I'm pretty new to apache so before I post my working code I wanted to double check if there is maybe something I missed:

1. I did not activate SSL in pyload .Would I have to, since the only external connection is the leg browser / apache. Are there potential risks here?

2. I had to write: 
ProxyPass http://localhost:8000 
ProxyPassReverse http://localhost:8000
instead of 
ProxyPass https://localhost:8000/pyload
ProxyPassReverse https://localhost:8000/pyload

I will post my full code working on Server version: Apache/2.4.18 (Ubuntu 16.04) + pyload.conf

Cheers
Julian
